crates/goose/src/agents/reference.rs

```diff
@@ -52,9 +52,20 @@ impl Agent for ReferenceAgent {
             .expect("Failed to list systems")
     }
 
-    async fn passthrough(&self, _system: &str, _request: Value) -> SystemResult<Value> {
-        // TODO implement
-        Ok(Value::Null)
+    async fn passthrough(&self, system: &str, method: &str, params: Value) -> SystemResult<Value> {
```

Comment:
meganit: maybe implement self.capabilities.passthrough helper - i think this code would be the same throughout